File: ruby/recursive_search.rb
def recursive_search(arr, target)
  return false if arr == []
  return true if arr[0] == target
  return false if arr.length == 1
  recursive_search(arr[1..-1], target)
end


# Please add your pseudocode to this file

# false if array is empty
# true if first number is target
# run search again with all but first num


# And a written explanation of your solution

# We are checking to see if the first element in the array 
# is the target element. If it's not, we're calling the function again
# with all but the first element. If the array is called when completely empty
# (after all other numbers have been eliminated as possible solutions), 
# the function returns false.

File: javascript/recursive_search.js
// const { SearchSource } = require('jest');

let index = 0;
function recursiveSearch(arr, target) {
  if (arr[index] === target) return true;
  console.log(arr[index]);
  index += 1;
  recursiveSearch(arr[index], target);
}

recursiveSearch([1, 2, 3], 2);

// if (require.main === module) {
//   // add your own tests in here
//   console.log('Expecting: true');
//   console.log('=>', recursiveSearch([1, 2, 3], 2));

//   console.log('');

//   console.log('Expecting: false');
//   console.log('=>', recursiveSearch([3, 2, 1], 4));
// }

// module.exports = recursiveSearch;

// Please add your pseudocode to this file
// if arr[i] = target: return
// search(i+1, target)

// And a written explanation of your solution
